src/util/recorder.ts

```diff
@@ -392,7 +392,9 @@ export class Recorder {
         // the abort is just for page, but download will succeed
         if (type === "Document" && reqresp.isValidBinary()) {
           this.serializeToWARC(reqresp);
-          //} else if (url) {
+          this.addPageRecord(reqresp);
+          this.removeReqResp(requestId);
+          return;
```

Comment:
Now that we have this return here, the else if statement below could be moved out of the conditional and the now unnecessary `break` removed.

Reply:
Hmm, there are other conditions around this, don't think that quite works? Removing the `break` results in lint error.

Reply:
Did change it to:
```
          this.removeReqResp(requestId);
          return this.serializeToWARC(reqresp);
```
which is consistent with other usage (and the addPageRecord is already part of serializeToWARC) and that way the promise from serializeToWARC can be awaited.

Reply:
cool, that works!